__tests__/app.test.js

```diff
@@ -195,6 +195,51 @@ describe("GET/api/articles/:article_id", () => {
     });
 })
 
+describe("GET/api/articles/:article_id?comment_count", () => {
+  test("200: gets an article by its article id and returns the object with the correct properties including comment_count when set to true", () => {
+    return request(app)
+      .get("/api/articles/1?comment_count=true")
+      .expect(200)
+      .then(({ body }) => {
+         const {article} = body
+         expect(typeof article.author).toBe("string")
+         expect(typeof article.title).toBe("string")
+         expect(typeof article.article_id).toBe("number")
+         expect(typeof article.body).toBe("string")
+         expect(typeof article.topic).toBe("string")
+         expect(typeof article.created_at).toBe("string")
+         expect(typeof article.votes).toBe("number")
+         expect(typeof article.author).toBe("string")
+         expect(typeof article.comment_count).toBe("number")
+      });
+  });
+  test("200: gets an article by its article id and returns the object with the correct properties excluding comment_count when set to false", () => {
+    return request(app)
+      .get("/api/articles/1?comment_count=false")
+      .expect(200)
+      .then(({ body }) => {
+         const {article} = body
+         expect(typeof article.author).toBe("string")
+         expect(typeof article.title).toBe("string")
+         expect(typeof article.article_id).toBe("number")
+         expect(typeof article.body).toBe("string")
+         expect(typeof article.topic).toBe("string")
+         expect(typeof article.created_at).toBe("string")
+         expect(typeof article.votes).toBe("number")
+         expect(typeof article.author).toBe("string")
+         expect(Object.keys(article).includes("comment_count")).toBe(false)
+      });
+  });
+  test("400: test that when given an invalid query (anything other than true or false) that bad request is returned", () => {
+      return request(app)
+        .get("/api/articles/1?comment_count=fasd")
+        .expect(400)
+        .then(({ body }) => {
+           expect(body.msg).toBe("Bad request")
+        });
+    });
```

Comment:
This should be just added to the get /api/articles/:article_id endpoint and should not be a query like you have done it here

Reply:
updated :)